src/par_iter/test.rs

```diff
@@ -656,6 +657,22 @@ pub fn par_iter_collect() {
 }
 
 #[test]
+pub fn par_iter_collect_vecdeque() {
+    let a: Vec<i32> = (0..1024).collect();
+    let b: VecDeque<i32> = a.par_iter().cloned().collect();
+    let c: VecDeque<i32> = a.iter().cloned().collect();
+    assert_eq!(b, c);
+}
+
+#[test]
+pub fn par_iter_collect_binaryheap() {
+    let a: Vec<i32> = (0..1024).collect();
+    let b: BinaryHeap<i32> = a.par_iter().cloned().collect();
+    assert_eq!(b.peek(), Some(&1023));
+    assert_eq!(b.len(), 1024);
```

Comment:
This is probably fine, but of course a loop that drains would be mildly better, no?

e.g.

```rust
for n in (0..1024).rev() {
    assert_eq!(b.pop(), Some(n));
    assert_eq!(b.len(), n);
}
```